src/Raven.Server/Web/System/DatabaseHelper.cs

```diff
@@ -95,6 +96,26 @@ public static void Validate(string name, DatabaseRecord record, RavenConfigurati
                         $"Problem trying to create a new sharded database {record.DatabaseName}."+
                         $" Sharded database can't have field {nameof(record.Topology)} defined in its record. Only the topologies inside {nameof(record.Sharding)} are relevant.");
                 }
+
+                var shardNodes = new HashSet<string>();
+                if (record.Sharding?.Shards?.Count > 0)
+                {
+                    foreach (var (shardNumber, topology) in record.Sharding.Shards)
+                    {
+                        if (topology?.Count > 0)
+                        {
+                            shardNodes.Clear();
+                            foreach (var node in topology.AllNodes)
+                            {
+                                if (shardNodes.Contains(node))
+                                    throw new InvalidOperationException(
+                                        $"Attempting to create sharded database {record.DatabaseName} but the topology of shard {shardNumber} contains node {node} more than once." +
+                                        " Can't have multiple replicas of the same shard on the same node.");
```

Comment:
This relevant to any database, not only for sharded.
Let's move this check to the `DatabaseTopology`

Reply:
Done